src/BaseVault.sol

```diff
@@ -674,7 +677,11 @@ abstract contract BaseVault is IVault, ERC20PermitUpgradeable, AccessControlUpgr
      * @dev This function iterates through the list of assets, gets their balances and rates,
      *      and updates the total assets denominated in the base asset.
      */
-    function processAccounting() public virtual {
+    function processAccounting() public virtual nonReentrant {
```

Comment:
add nonReentrant as well so an evil token cannot reenter through processAccounting while _withdraw is happening